src/components/Operation.jsx

```diff
@@ -0,0 +1,122 @@
+import { faArrowLeft } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Component } from 'react'
+import { Link } from 'react-router-dom'
+// import { Link } from 'react-router-dom'
+import './style/operation.css'
+import Snackbar from './subComponents/Snackbar'
+class Operation extends Component {
+    constructor() {
+        super()
+        this.state = {
+            amount: 0,
+            vendor: "",
+            category: "",
+            date: Date.now(),
+            errorMessage: "",
+        }
+        this.timer = {}
+    }
+
+    handleInput = (e) => {
+        const state = { ...this.state }
+        const value = e.target.type === "number" ? parseInt(e.target.value) : e.target.value
+        const name = e.target.name
+        state[name] = value
+        this.setState(state)
+    }
+    
+    toProperCase = (str)=>{
+        const arr = str.split('')
+        const firstLetter = arr[0].toUpperCase()
+        arr.splice(0,1)
+        return firstLetter + arr.join('').toLowerCase()
+    }
+
+    emptyInput = () => {
+        this.setState({
+            amount: 0,
+            vendor: "",
+            category: "",
+            date: Date.now()
+        })
+    }
+
+    resetMsg = () => {
+
+        this.setState({
+            errorMessage: ""
+        })
+    }
+
+    showMsg = (message) => {
+        this.setState({
+            errorMessage: ""
+        }, () => {
+
+            this.setState({
+                errorMessage: message
+            })
+        })
+    }
+
+    transaction = async (type) => {
```

Comment:
the name of this method should be understandable for the reader - `transaction` is not such a good name...